src/audio/pipeline/pipeline-stream.c

```diff
@@ -47,6 +47,17 @@ pipeline_should_report_enodata_on_trigger(struct comp_dev *rsrc,
 	struct pipeline_data *ppl_data = ctx->comp_data;
 	struct comp_dev *pipe_source = ppl_data->start->pipeline->source_comp;
 
+	/* In IPC3, FW propagates triggers to connected pipelines, so
+	 * it can have determistic logic to conclude no data is
+	 * available.
+	 * In IPC4, host controls state of each pipeline separately,
+	 * so FW cannot reliably detect case of no data based on
+	 * observing state of src->pipeline here.
+	 */
+#if CONFIG_IPC_MAJOR_4
+	return false;
+#endif
```

Comment:
nitpick / not a blocker: this is a bit crude too... We usually enclose the whole function in that preprocessor conditional bracket. I know I'm too late, approving anyway